### DISCUSSION TITLE
Getting the langium web worker example setup with react

### DISCUSSION THREAD

**Discussion by SorenHolstHansen** [TypeFox/monaco-languageclient#481] [Q&A]
I was trying my way with a very basic react setup with web workers. I tried looking at the langium example and the react example and combining the two, while also using @monaco-editor/react.

[The code](https://github.com/SorenHolstHansen/monaco-react-languageclient) compiles fine, but there is no auto-completion, and I can see in the console, that the web workers have not been setup correctly.

I guess this is not strictly related to monaco-languageserver, but I was hoping that I could get some help from you to make it work.

My ultimate goal was to get the monaco editor working with the [dbt lsp](https://github.com/fivetran/dbt-language-server/tree/main) with a web worker, but I wanted to try a basic example first to see if I could get it to work.

Thanks for any help

**Comment by kaisalmen** [comment]
Hi @SorenHolstHansen I will take a look at your repo, try it and see if I can provide with an idea for a way forward.

**Reply by SorenHolstHansen** [reply]
Thanks a lot!

**Reply by kaisalmen** [reply]
There are two things I see. **react-scripts** does not load the editor-worker. It shows HTML ⬇️ . Don't know why and I don't know react-scripts?
![image](https://user-images.githubusercontent.com/21079138/231799974-410ebec2-3a3b-4e05-9e49-7e1d8389f961.png)
⬆️ I ran this on Windows

You miss the worker build step from the langium example. The package.json contains this step during build:
https://github.com/TypeFox/monaco-languageclient/blob/main/packages/examples/main/package.json#L41

Therefore the worker cannot be loaded. Hope this helps.

**Reply by kaisalmen** [reply]
Same behaviour on Linux, btw

**Reply by SorenHolstHansen** [reply]
Thanks a lot! I will give it another shot
